File: backend/Entities/Tournament.cs
using backend.Enums;

namespace backend.Entities;

public class Tournament
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public TournamentStatus Status { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public string Region { get; set; } = null!;
    public string City { get; set; } = null!;
    public string Country { get; set; } = null!;
    public decimal PrizePool { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: frontend/src/types/Tournament.ts
export type TournamentStatus = 'Live' | 'Finished' | 'Upcoming';

export type TournamentFilter = TournamentStatus | 'All';

export type TournamentOption = {
  id: string | number;
  name: string;
};

export interface Tournament {
  id: number;
  name: string;
  status: TournamentStatus;
  startDate: string;
  endDate: string;
  region: string;
  prizePool: number;
  teamCount: number;
}
